pkg/chains/formats/slsa/v2/taskrun/taskrun.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2022 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package taskrun
+
+import (
+	"context"
+	"reflect"
+
+	intoto "github.com/in-toto/in-toto-golang/in_toto"
+	slsa "github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/v0.2"
+	"github.com/tektoncd/chains/pkg/chains/formats/slsa/extract"
+	slsav1 "github.com/tektoncd/chains/pkg/chains/formats/slsa/v1/taskrun"
+	"github.com/tektoncd/chains/pkg/chains/objects"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"knative.dev/pkg/logging"
+)
+
+// BuildConfig is the custom Chains format to fill out the
+// "buildConfig" section of the slsa-provenance predicate
+type BuildConfig struct {
+	TaskSpec *v1beta1.TaskSpec `json:"taskSpec"`
+}
+
+func GenerateAttestation(builderID string, tro *objects.TaskRunObject, ctx context.Context) (interface{}, error) {
+	logger := logging.FromContext(ctx)
+	subjects := extract.SubjectDigests(tro, logger)
+
+	mat, err := slsav1.Materials(tro, logger)
+	if err != nil {
+		return nil, err
+	}
+	att := intoto.ProvenanceStatement{
+		StatementHeader: intoto.StatementHeader{
+			Type:          intoto.StatementInTotoV01,
+			PredicateType: slsa.PredicateSLSAProvenance,
+			Subject:       subjects,
+		},
+		Predicate: slsa.ProvenancePredicate{
+			Builder: slsa.ProvenanceBuilder{
+				ID: builderID,
+			},
+			BuildType:   tro.GetGVK(),
```

Comment:
We'll need a different BuildType here so that clients can detect whether they are dealing with v1 or v2 types.

I'm thinking something along the lines of `https://chains.tekton.dev/format/<format type>/type/<gvk>` - i.e. `https://chains.tekton.dev/format/slsa/v2/type/tekton.dev/v1beta1/TaskRun` - wdyt?

Reply:
That looks good! How do we populate that link with information? May be that might eventually lead to the buildType templating that Christie was talking about.